src/main/java/com/gmail/yuramitryahin/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.gmail.yuramitryahin.dao;
+
+import com.gmail.yuramitryahin.exception.DataProcessingException;
+import com.gmail.yuramitryahin.lib.Dao;
+import com.gmail.yuramitryahin.model.Movie;
+import com.gmail.yuramitryahin.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(id);
```

Comment:
```suggestion
            transaction.commit();
```
You don’t need to set id for your entity explicitly it’ll be done by Hibernate ([Link](https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html))

Reply:
fixed